extension/src/connect/index.ts

```diff
@@ -0,0 +1,118 @@
+import { commands, ExtensionContext, SecretStorage } from 'vscode'
+import { validateTokenInput } from './inputBox'
+import { STUDIO_ACCESS_TOKEN_KEY, isStudioAccessToken } from './token'
+import { STUDIO_URL } from './webview/contract'
+import { Resource } from '../resourceLocator'
+import { ViewKey } from '../webview/constants'
+import { MessageFromWebview, MessageFromWebviewType } from '../webview/contract'
+import { BaseRepository } from '../webview/repository'
+import { Logger } from '../common/logger'
+import { getInput, getValidInput } from '../vscode/inputBox'
+import { Title } from '../vscode/title'
+import { openUrl } from '../vscode/external'
+import { ContextKey, setContextValue } from '../vscode/context'
+import { RegisteredCommands } from '../commands/external'
+
+export class Connect extends BaseRepository<undefined> {
+  public readonly viewKey = ViewKey.CONNECT
+
+  private readonly secrets: SecretStorage
+
+  constructor(context: ExtensionContext, webviewIcon: Resource) {
+    super('', webviewIcon)
+
+    this.secrets = context.secrets
+
+    this.dispose.track(
+      this.onDidReceivedWebviewMessage(message =>
+        this.handleMessageFromWebview(message)
+      )
+    )
+
+    void this.setContext().then(() => this.deferred.resolve())
+
+    this.dispose.track(
+      context.secrets.onDidChange(e => {
+        if (e.key !== STUDIO_ACCESS_TOKEN_KEY) {
+          return
+        }
+        return this.setContext()
+      })
+    )
+  }
+
+  public sendInitialWebviewData(): void {}
+
+  public removeStudioAccessToken() {
+    return this.removeSecret(STUDIO_ACCESS_TOKEN_KEY)
+  }
+
+  public async saveStudioAccessToken() {
+    const token = await getValidInput(
+      Title.ENTER_STUDIO_TOKEN,
+      validateTokenInput,
+      { password: true }
+    )
+    if (!token) {
+      return
+    }
+
+    return this.storeSecret(STUDIO_ACCESS_TOKEN_KEY, token)
```

Comment:
Should we add some kind of message after the token is saved? Feels a little weird to me that I saved my token then everything related to Studio just disappears from my screen 🤔

Reply:
Yes I will add a toast/modal

Reply:
This is what the flow looks like now: 

https://user-images.githubusercontent.com/37993418/219277549-fc5af759-06e5-4682-ada1-80eda3c77207.mov

Reply:
Let me know what you think. We can update in a follow up.

Reply:
Looks good! Makes things more clear on the connection and what you can do now that you're connected!